src/main/java/ognl/ObjectPropertyAccessor.java

```diff
@@ -97,13 +97,13 @@ public Object setPossibleProperty(Map context, Object target, String name, Objec
         try {
             if (!OgnlRuntime.setMethodValue(ognlContext, target, name, value, true))
             {
-                result = OgnlRuntime.setFieldValue(ognlContext, target, name, value) ? null : OgnlRuntime.NotFound;
+                result = OgnlRuntime.setFieldValue(ognlContext, target, name, value, true) ? null : OgnlRuntime.NotFound;
             }
 
             if (result == OgnlRuntime.NotFound)
             {
                 Method m = OgnlRuntime.getWriteMethod(target.getClass(), name);
-                if (m != null)
+                if (m != null && ognlContext.getMemberAccess().isAccessible(context, target, m, name))
```

Comment:
This is a new bug discovered by @jefferyxhy which is much worse, it allows public setters on excluded classes to be called!